sdk/identity/azure-identity/src/main/java/com/azure/identity/AppServiceMsiCredential.java

```diff
@@ -30,13 +32,24 @@
      */
     AppServiceMsiCredential(String clientId, IdentityClient identityClient) {
         Configuration configuration = Configuration.getGlobalConfiguration().clone();
+        this.identityEndpoint = configuration.get(ManagedIdentityCredential.PROPERTY_IDENTITY_ENDPOINT);
+        this.identityHeader = configuration.get(ManagedIdentityCredential.PROPERTY_IDENTITY_HEADER);
         this.msiEndpoint = configuration.get(Configuration.PROPERTY_MSI_ENDPOINT);
         this.msiSecret = configuration.get(Configuration.PROPERTY_MSI_SECRET);
         this.identityClient = identityClient;
         this.clientId = clientId;
-        if (!(msiEndpoint.startsWith("https") || msiEndpoint.startsWith("http"))) {
+        if (identityEndpoint != null) {
+            validateEndpointProtocol(this.identityEndpoint, "Identity");
+        } else {
+            validateEndpointProtocol(this.msiEndpoint, "MSI");
+        }
```

Comment:
What happens when both identity and msi endpoints are set? Is there a priority order for which endpoint to use? Is it documented somewhere? Also, if both are set, we should validate them both.

Reply:
if identity endpoint is set, we just use that, else we use MSI endpoint if its set.
Although we use only identity endpoint, if both are set, but for validation, I have enabled it for both if both are set.